tests/MatchesSnapshotTest.php

```diff
@@ -0,0 +1,223 @@
+
+        $mockTrait
+            ->expects($this->once())
+            ->method('assertJsonStringEqualsJsonString')
+            ->willThrowException(new PHPUnit_Framework_ExpectationFailedException(''));
+
+        $this->expectException(PHPUnit_Framework_ExpectationFailedException::class);
+
+        $mockTrait->assertMatchesJsonSnapshot('{"foo":"foo","bar":"bar","baz":"baz"}');
+    }
+
+    /** @test */
+    public function it_can_update_a_string_snapshot()
+    {
+        $_SERVER['argv']['test'] = '--update';
+
+        $mockTrait = $this->getMatchesSnapshotMock();
+
+        $this->expectIncompleteMatchesSnapshotTest($mockTrait);
+
+        $mockTrait->assertMatchesSnapshot('Bar');
+
+        $snapshot = $this->snapshotsDir.DIRECTORY_SEPARATOR.'MatchesSnapshotTest__it_can_update_a_string_snapshot.php';
+
+        $this->assertStringEqualsFile($snapshot, "<?php return 'Bar';");
+    }
+
+    /** @test */
+    public function it_can_update_a_xml_snapshot()
+    {
+        $_SERVER['argv'][] = '--update';
+
+        $mockTrait = $this->getMatchesSnapshotMock();
+
+        $this->expectIncompleteMatchesSnapshotTest($mockTrait);
+
+        $mockTrait->assertMatchesXmlSnapshot('<foo><bar>Baz</bar></foo>');
+
+        $snapshot = $this->snapshotsDir.DIRECTORY_SEPARATOR.'MatchesSnapshotTest__it_can_update_a_xml_snapshot.xml';
+
+        $this->assertFileEquals($this->exampleSnapshotsDir.DIRECTORY_SEPARATOR.'snapshot.xml', $snapshot);
+    }
+
+    /** @test */
+    public function it_can_update_a_json_snapshot()
+    {
+        $_SERVER['argv'][] = '--update';
+
+        $mockTrait = $this->getMatchesSnapshotMock();
+
+        $this->expectIncompleteMatchesSnapshotTest($mockTrait);
+
+        $mockTrait->assertMatchesJsonSnapshot('{"foo":"foo","bar":"bar","baz":"baz"}');
+
+        $snapshot = $this->snapshotsDir.DIRECTORY_SEPARATOR.'MatchesSnapshotTest__it_can_update_a_json_snapshot.json';
+
+        $this->assertFileEquals($this->exampleSnapshotsDir.DIRECTORY_SEPARATOR.'snapshot.json', $snapshot);
+    }
+
+    protected function expectIncompleteMatchesSnapshotTest(PHPUnit_Framework_MockObject_MockObject $matchesSnapshotMock)
+    {
+        $matchesSnapshotMock
+            ->expects($this->atMost(1))
+            ->method('assertEquals')
+            ->willThrowException(new PHPUnit_Framework_ExpectationFailedException(''));
+
+        $matchesSnapshotMock
+            ->expects($this->atMost(1))
+            ->method('assertXmlStringEqualsXmlString')
+            ->willThrowException(new PHPUnit_Framework_ExpectationFailedException(''));
+
+        $matchesSnapshotMock
+            ->expects($this->atMost(1))
+            ->method('assertJsonStringEqualsJsonString')
+            ->willThrowException(new PHPUnit_Framework_ExpectationFailedException(''));
+
+        $matchesSnapshotMock
+            ->expects($this->once())
+            ->method('markTestIncomplete');
+    }
+
+    /**
+     * @return PHPUnit_Framework_MockObject_MockObject
```

Comment:
Not sure what would be more appropriate here. `getMockForTrait` returns an instance of a randomly named class (for example `Mock_Trait_MatchesSnapshots_7d22af5e_56383e1b#1329`). That class implements the `PHPUnit_Framework_MockObject_MockObject` interface.